pkg/handler/metatile.go

```diff
@@ -86,134 +92,245 @@ func MetatileHandler(
 			}
 		}
 
-		metatileData := parseResult.AdditionalData.(*MetatileParseData)
+		metatileData := parseResult.AdditionalData.(*state.MetatileParseData)
 		reqState.Coord = &metatileData.Coord
 		reqState.Format = reqState.Coord.Format
 		reqState.HttpData = parseResult.HttpData
 
-		metaCoord, offset, err := metatileData.Coord.MetaAndOffset(metatileSize, tileSize, metatileMaxDetailZoom)
-		if err != nil {
-			logger.Warning(log.LogCategory_ConfigError, "MetaAndOffset could not be calculated: %s", err.Error())
-			http.Error(rw, err.Error(), http.StatusInternalServerError)
-			reqState.ResponseState = state.ResponseState_Error
-			// Note: FetchState is left as nil, since no fetch was performed
-			return
+		if tileCache == nil {
```

Comment:
Yea, I did. This snuck in because the tests call this directly. I'm thinking this should stay here and I remove the server.go version. That way this protects itself against inadvertently setting a nil cache and getting a null pointer.

Reply:
Fixed in c91602d.